ethereum/migrations/test/4_pyth_multiple_emitters.js

```diff
@@ -0,0 +1,13 @@
+require('dotenv').config({ path: "../.env" });
+
+const PythUpgradable = artifacts.require("PythUpgradable");
+
+const { upgradeProxy } = require("@openzeppelin/truffle-upgrades");
+
+/**
+ * Adds multiple emitter/chain ID support
+ */
+module.exports = async function (deployer) {
+    const instance = await PythUpgradable.deployed();
+    await upgradeProxy(instance.address, PythUpgradable, { deployer, call: "migrateMultiSources" });
```

Comment:
Will this migration function be even called if the contract has not changed?

Reply:
I think Truffle tracks which migrations were already applied. Ideally, prod will use this only once. @tompntn is that right?

Reply:
yeah it tracks migration numbers in a separate contract. But my question is different,

If I go and deploy from scratch in xyz network then from the first deployment contract is the latest version. Then truffle gets to this migration file and I know that truffle won't upload a new implementation when the contract is already updated. My question is that does it call this function either-way? 

After some thought I believe the answer is yes because tilt is working and tilt testing scenario is exactly as I said.